src/skins/vector/views/molecules/MNoticeTile.js

```diff
@@ -17,18 +17,67 @@ limitations under the License.
 'use strict';
 
 var React = require('react');
+var sanitizeHtml = require('sanitize-html');
 
 var MNoticeTileController = require('matrix-react-sdk/lib/controllers/molecules/MNoticeTile')
 
+var allowedAttributes = sanitizeHtml.defaults.allowedAttributes;
+allowedAttributes['font'] = ['color'];
+var sanitizeHtmlParams = {
+    allowedTags: sanitizeHtml.defaults.allowedTags.concat([ 'font' ]),
+    allowedAttributes: allowedAttributes,
+};
+
 module.exports = React.createClass({
     displayName: 'MNoticeTile',
     mixins: [MNoticeTileController],
 
+    // FIXME: this entire class is copy-pasted from MTextTile :(        
     render: function() {
         var content = this.props.mxEvent.getContent();
+        var body = content.body;
+
+        if (content.format === "org.matrix.custom.html") {
+            body = sanitizeHtml(content.formatted_body, sanitizeHtmlParams);
+        }
+
+        if (this.props.searchTerm) {
+            var lastOffset = 0;
+            var bodyList = [];
+            var k = 0;
+            var offset;
+            // XXX: this probably doesn't handle stemming very well.
+            while ((offset = body.indexOf(this.props.searchTerm, lastOffset)) >= 0) {
+                if (content.format === "org.matrix.custom.html") {
+                    // FIXME: we need to apply the search highlighting to only the text elements of HTML, which means
+                    // hooking into the sanitizer parser rather than treating it as a string.  Otherwise
+                    // the act of highlighting a <b/> or whatever will break the HTML badly.
+                    bodyList.push(<span key={ k++ } dangerouslySetInnerHTML={{ __html: body.substring(lastOffset, offset) }} />);
```

Comment:
XSS here by setting HTML in `event.body`. This is not protected by `:41` as that is only applied if the `format` key has the html type.